### PR TITLE
Add terrain_map_publisher to spirit_utils

### PR DESCRIPTION
**Pull request by jcnorby**
And implement in planning examples.

### CODE REVIEW

File: spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#include <ros/ros.h>
+#include <sensor_msgs/Image.h>
+
+#include <grid_map_ros/grid_map_ros.hpp>
+#include "grid_map_core/GridMap.hpp"
+#include "grid_map_core/TypeDefs.hpp"
+
+#include <string>
+
+class ImageToGridmapDemo
```
Comment (nikolaif399) [inline spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp:22]: Class doxy brief required

File: spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  /*!
+  * Reads and verifies the ROS parameters.
+  * @return true if successful.
+  */
+  bool readParameters();
+
+  void imageCallback(const sensor_msgs::Image& msg);
+
+  grid_map::GridMap dataSparseMap_;
```
Comment (nikolaif399) [inline spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp:45]: Doxygen

File: spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  //! Length of the grid map in x direction.
+  double mapLengthX_;
+
+  //! Resolution of the grid map.
+  double resolution_;
+
+  //! Range of the height values.
+  double minHeight_;
+  double maxHeight_;
```
Comment (nikolaif399) [inline spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp:73]: Separate into two doxy comments (one per var)

File: spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  //! Frame id of the grid map.
+  std::string mapFrameId_;
+
+  bool mapInitialized_;
```
Comment (nikolaif399) [inline spirit_utils/include/spirit_utils/ImageToGridmapDemoCustom.hpp:78]: Doxy

**Review by nikolaif399 — CHANGES_REQUESTED** [review]

**Comment by nikolaif399** [comment]
Need some doxygen comments on the ImageToGridMap class, otherwise looks good. I see you fixed the build warning about depending on grid_map directly.

**Comment by jcnorby** [comment]
Oh, whoops, good catch - ImageToGridMap was supposed to be deleted, I merged it into the terrain_map_publisher class. Fixed, and updated the yaml and config files as well.

**Review by jcnorby — COMMENTED** [review]
Can you take a look at this and approve if good?

**Review by nikolaif399 — APPROVED** [review]

**Pull request merged**